pkg/operator/ceph/controller/spec.go

```diff
@@ -75,21 +75,64 @@ var (
 	cronLogRotate = `
 CEPH_CLIENT_ID=%s
 PERIODICITY=%s
-LOG_ROTATE_CEPH_FILE=/etc/logrotate.d/ceph
+LOG_ROTATE_CEPH_FOLDER=/etc/logrotate.d/ceph
+DAEMON_LOG_FILE=/var/log/ceph/
+LOG_MAX_SIZE=%s
+
+# size of log after removing any alphabet in the end. eg: for '1G' getting '1'
+LOG_MAX_SIZE_IN_DIGIT=$(echo "${LOG_MAX_SIZE}" | sed 's/.$//')
+# get last char of file size. eg: for '1G' getting 'G'
+GET_LAST_CHAR_OF_LOG_SIZE=$(echo $LOG_MAX_SIZE | rev | cut -c1)
+
+if [ "$GET_LAST_CHAR_OF_LOG_SIZE" == "G" ]; then
+	LOG_MAX_SIZE_IN_KB=$(( (LOG_MAX_SIZE_IN_DIGIT *1024*1024)))
+elif [ "$GET_LAST_CHAR_OF_LOG_SIZE" == "M" ]; then
+	LOG_MAX_SIZE_IN_KB=$(( (LOG_MAX_SIZE_IN_DIGIT *1024)))
+fi
+
+LAST_ROTATION_TIME_IN_SEC=$(echo $SECONDS)
 
 if [ -z "$PERIODICITY" ]; then
 	PERIODICITY=24h
 fi
 
+# remove last char from PERIODICITY eg. for 24h getting 24.
+PERIODICITY_IN_DIGIT=$(echo $PERIODICITY | awk '{print $1}' | sed 's/.$//')
+
 # edit the logrotate file to only rotate a specific daemon log
 # otherwise we will logrotate log files without reloading certain daemons
 # this might happen when multiple daemons run on the same machine
-sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FILE"
+sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FOLDER"
 
 while true; do
-	sleep "$PERIODICITY"
-	echo "starting log rotation"
-	logrotate --verbose --force "$LOG_ROTATE_CEPH_FILE"
+	SIZE_OF_FILE=$(du --summarize --human-readable $DAEMON_LOG_FILE | awk '{print $1}')
+
+	# size of file after removing any alphabet in the end. eg: for '1G' getting '1'
+	SIZE_OF_FILE_IN_DIGIT=$(du --summarize --human-readable $DAEMON_LOG_FILE | awk '{print $1}' | sed 's/.$//')
+	# get last char of file size. eg: for '1G' getting 'G'
+	GET_LAST_CHAR_OF_FILE_SIZE=$(echo $SIZE_OF_FILE | rev | cut -c1)
+
+	if [ "$GET_LAST_CHAR_OF_FILE_SIZE" != "G" ] && [ "$GET_LAST_CHAR_OF_FILE_SIZE" != "M" ]; then
+		SIZE_OF_FILE_IN_KB=0
+	elif [ "$GET_LAST_CHAR_OF_FILE_SIZE" == "G" ]; then
+		SIZE_OF_FILE_IN_KB=$(echo $SIZE_OF_FILE_IN_DIGIT *1024 | awk '{print $1*1024}'| cut -d $(locale decimal_point) -f 1)
+	elif [ "$GET_LAST_CHAR_OF_FILE_SIZE" == "M" ]; then
+		SIZE_OF_FILE_IN_KB=$(echo $SIZE_OF_FILE_IN_DIGIT *1024*1024 | awk '{print $1*1024}'| cut -d $(locale decimal_point) -f 1)
+	fi
+
+	CURRENT_TIME_IN_SEC=$(echo $SECONDS)
+	TIME_DIFF_IN_HOUR=$(( (LAST_ROTATION_TIME_IN_SEC - CURRENT_TIME_IN_SEC) / 3600 ))
+	if [ "$TIME_DIFF_IN_HOUR" -lt 0 ]; then
+		TIME_DIFF_IN_HOUR=$((-TIME_DIFF_IN_HOUR))
+	fi
+
+	if [[ $TIME_DIFF_IN_HOUR -ge $PERIODICITY_IN_DIGIT ]] || [[ $SIZE_OF_FILE_IN_KB -ge $LOG_MAX_SIZE_IN_KB ]]; then
+		echo "starting log rotation"
+		logrotate --verbose --force "$LOG_ROTATE_CEPH_FOLDER"
```

Comment:
To clarify earlier discussion about the log folder, size, and file(s) to rotate, I believe we want to only rotate the single file here. Since each daemon pod has its own sidecar container that will rotate the logs for that individual daemon, we should only detect the size of that one log file, and only rotate that one log file in this script. 
```suggestion
		logrotate --verbose --force "$LOG_ROTATE_CEPH_FILE"
```